### ISSUE TITLE
Add `__toString` on ApiResponse

### ISSUE THREAD

**Issue by dextermb** [langleyfoxall/helpers-laravel#11]
I think that we should add `__toString()` to `ApiResponse` as when returning from a controller it attempts to convert the returned object to a string, if we were to implement this it would save us from adding `->json()` on the end of every API return.

_Proposed code_
```
class ApiResponse
{
    // ...

    public function __toString()
    {
        return $this->json()
    }
```

Thoughts? We could also implement [`ArrayAccess`](http://php.net/manual/en/class.arrayaccess.php) for even more accessibility.

cc @DivineOmega @JacobBrassington @wheatleyjj

**Comment by DivineOmega** [comment]
+1 on implementing `__toString()` in this manner.

I'm curious about your desired implementation of the `ArrayAccess` interface. Maybe worth making a new issue for this?

**Comment by dextermb** [comment]
@DivineOmega Nothing crazy, would just allow people to push into `data`. Maybe something like the example below

```
$initial_data = ['message' => 'hello, world'];
$response = ApiResponse::success() # or new ApiResponse or ApiResponse::error()

$response->data($initial_data);

// Edit array elements
if(false) {
    $response['message'] = 'foo bar';
}

// Add new elements to data
if(true) {
    $response['users'] = [];
}

// or just $response with __toString
return $response->json();

// Expected response after transforming data along the way
//
// {
//     "status": 200,
//     "success": true,
//     "error": null,
//     "data": {
//         'message': 'hello, world',
//         'users': []
//     }
//     "meta": null
// }
```

**Comment by DivineOmega** [comment]
Looks good. Shouldn't be breaking in any way.

**Comment by dextermb** [comment]
@DivineOmega Just noticed an issue, `__toString` must return a string... obviously but a response class is not an string so `->json()` wouldn't work.

**Comment by DivineOmega** [comment]
@dextermb In that case, the best thing you could do is to make `__toString` return `json_encode(get_object_vars($this));`. It would be missing the HTTP status code, and the JSON header, though.

**Comment by dextermb** [comment]
@DivineOmega Would Laravel's response pick it up as JSON and know to respond with JSON headers?

**Comment by DivineOmega** [comment]
@dextermb No, it would return a plain text response unless to requester specifically asked for JSON.

Idea: Why not extend `Illuminate\Http\Response`? The class would then be a proper response itself, so there would be no need for this manual conversion.

**Comment by DivineOmega** [comment]
@dextermb Or perhaps even better, just implement `Illuminate\Contracts\Support\Renderable` and create a `render()` method that calls and returns your `json()` method.

**Comment by dextermb** [comment]
Would like to see someone have a go at this.
